src/app/settings/settings.page.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, OnInit } from '@angular/core';
+import { ModalController, ToastController } from '@ionic/angular';
+import { UserService, Preferences, VoiceActor, VoiceActorService, PresentationOrder } from 'wanikani-api-ng';
+import { Observable } from 'rxjs';
+import { map, tap, take } from 'rxjs/operators';
+import { FormGroup, FormControl } from '@angular/forms';
+
+@Component({
+  selector: 'app-settings',
+  templateUrl: './settings.page.html',
+  styleUrls: ['./settings.page.scss'],
+})
+export class SettingsPage implements OnInit {
+  preferences: Observable<Preferences>
+  voiceActors: Observable<VoiceActor[]>
+  //WaniKani accepts batch sizes from 3 to 10 inclusive this array is for using ngFor
+  batchSizes = Array(8).fill(1).map((x,i)=>i+3)
+
+  //these are used for iterating over enums in ngFor
+  keys = Object.keys
+  presentationOrderEnum = PresentationOrder
+
+  preferencesForm = new FormGroup({
+    default_voice_actor_id: new FormControl(""),
+    lessons_autoplay_audio: new FormControl(""),
+    lessons_batch_size: new FormControl(""),
+    lessons_presentation_order: new FormControl(""),
+    reviews_autoplay_audio: new FormControl(""),
+    reviews_display_srs_indicator: new FormControl(""),
+  })
+
+  constructor(  private modalController: ModalController, 
+                private userService: UserService, 
+                private voiceActorService: VoiceActorService,
+                private toastController: ToastController) { }
+
+  ngOnInit() {
+    this.preferences = this.userService.getUser().pipe(
+      map(user => user.data.preferences),
+      take(1) //prevents the following subscribe from triggering more than once?
```

Comment:
We should takeUntil this component gets destroyed instead of taking 1.

Reply:
Do you have an example of this? Specifically the notifier part of the `takeUntil()` I believe if we go this route then the subscribe will be called with every preference update so I can remove the `patchValue` from the subscribe in `updatePreferences()`

Reply:
create a subject called destroyed `destroyed = new Subject()` the type doesn't matter. in your pipe add `takeUntil(this.destroyed)`. Then implement OnDestroy and in the ngOnDestroy method call next on destroyed `this.destroyed.next()`. And you're right about the subscribe callback being called on update. The observable you are subscribing to will get the value returned from the update preferences put request.

Reply:
This should all be updated